### PR TITLE
PYTHON-3290 Support nested pymongo.timeout() calls

### PR DESCRIPTION
**Pull request by ShaneHarvey**
https://jira.mongodb.org/browse/PYTHON-3290

### CODE REVIEW

File: pymongo/__init__.py [hunk 1]
```diff
@@ -127,6 +127,18 @@ def timeout(seconds: Optional[float]) -> ContextManager:
...
           print(f"block timed out: {exc!r}")
 
+    When nesting :func:`~pymongo.timeout`, the nested block overrides the
+    timeout. When exiting the block, the previous deadline is restored::
+
+      with pymongo.timeout(5):
+          coll.find_one()  # Uses the 5 second deadline.
+          with pymongo.timeout(3):
+              coll.find_one() # Uses the 3 second deadline.
+          coll.find_one()  # Uses the 5 second deadline.
```
Comment (blink1073) [inline pymongo/__init__.py:137]: Perhaps clarify that this is the *original* 5 second deadline, not a new one.

Comment (ShaneHarvey) [reply pymongo/__init__.py]: Done.

**Comment by ShaneHarvey** [comment]
Ready for another look.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**